frontend/packages/dev-console/src/components/import/deployImage-submit-utils.ts

```diff
@@ -0,0 +1,287 @@
+        ...(triggers.image ? [{ type: 'ImageChange', imageChange: {} }] : []),
+        ...(triggers.config ? [{ type: 'ConfigChange' }] : []),
+      ],
+    },
+  };
+
+  return k8sCreate(BuildConfigModel, buildConfig, dryRun ? dryRunOpt : {});
+};
+
+export const createDeploymentConfig = (
+  formData: DeployImageFormData,
+  dryRun: boolean,
+): Promise<K8sResourceKind> => {
+  const {
+    project: { name: namespace },
+    application: { name: application },
+    name,
+    searchTerm,
+    isi: { name: isi_name, tag, ports },
+    deployment: { env, replicas, triggers },
+    labels: userLabels,
+  } = formData;
+
+  const defaultLabels = getAppLabels(name, application, isi_name);
+  const labels = _.isEmpty(userLabels) ? {app: application} : SelectorInput.objectify(userLabels);
+
+  const deploymentConfig = {
+    kind: 'DeploymentConfig',
+    apiVersion: 'apps.openshift.io/v1',
+    metadata: {
+      name,
+      namespace,
+      labels: { ...defaultLabels, ...userLabels },
+      annotations,
+    },
+    spec: {
+      replicas,
+      selector: labels,
+      template: {
+        metadata: {
+          labels,
+          annotations,
+        },
+        spec: {
+          volumes,
+          containers: [
+            {
+              name,
+              image: `${searchTerm}:latest`,
+              ports,
+              volumeMounts,
+              env,
+            },
+          ],
+        },
+      },
+      triggers: [
+        {
+          type: 'ImageChange',
+          imageChangeParams: {
+            automatic: triggers.image,
+            containerNames: [name],
+            from: {
+              kind: 'ImageStreamTag',
+              name: `${name}:${tag}`,
+            },
+          },
+        },
+        ...(triggers.config ? [{ type: 'ConfigChange' }] : []),
+      ],
+    },
+  };
+
+  return k8sCreate(DeploymentConfigModel, deploymentConfig, dryRun ? dryRunOpt : {});
+};
+
+export const createService = (
+  formData: DeployImageFormData,
+  dryRun: boolean,
+): Promise<K8sResourceKind> => {
+  const {
+    project: { name: namespace },
+    application: { name: application },
+    name,
+    isi: { name: isi_name, ports },
+    labels: userLabels,
+  } = formData;
+
+  const defaultLabels = getAppLabels(name, application, isi_name);
+  const podLabels = getPodLabels(name);
+  const service = {
+    kind: 'Service',
+    apiVersion: 'v1',
+    metadata: {
+      name,
+      namespace,
+      labels: { ...defaultLabels, ...userLabels },
+      annotations,
+    },
+    spec: {
+      selector: podLabels,
+      ports: _.map(ports, (port) => ({
+        port: port.containerPort,
+        targetPort: port.containerPort,
+        protocol: port.protocol,
+        // Use the same naming convention as CLI new-app.
+        name: `${port.containerPort}-${port.protocol}`.toLowerCase(),
+      })),
+    },
+  };
+
+  return k8sCreate(ServiceModel, service, dryRun ? dryRunOpt : {});
+};
+
+export const createRoute = (
```

Comment:
Is it intentional that this is different than git import? There's no hostname set. Can we share he logic if they are supposed to be the same.

Reply:
They get the image from different places. That is why they have to be different.

Reply:
That might be possible but at this point I would rather leave it for a later refactoring.